fontforge/python.c

```diff
@@ -19744,9 +19743,26 @@ void FontForge_InitializeEmbeddedPython(void) {
     if ( python_initialized )
 	return;
 
-    SetPythonProgramName("fontforge");
+    PyConfig config;
+    PyStatus status;
+    PyConfig_InitPythonConfig(&config);
```

Comment:
Since this is an embedded usecase, should this be [`PyConfig_InitIsolatedConfig`](https://docs.python.org/3/c-api/init_config.html#c.PyConfig_InitIsolatedConfig) instead?

Reply:
It depends on the behavior that you want. Isolated mode ignores env vars for example. I used Python mode to avoid behavior changes, only to use the new API.

Reply:
Yes, not changing the behavior in a refactor makes a lot of sense.

I think there's an argument to changing from this current behavior, too, since it doesn't make too much sense to me to prioritize the current working directory to load (say) `pkg_resources` over what's in the system path, yet that's what's done today. (The first entry in `sys.path` is the empty string, which loads from the current directory.)

https://github.com/fontforge/fontforge/blob/9af60edefc61d1f9244601802067c33fd221db69/fontforge/plugin.c#L293-L295

The current behavior has shades of [DLL hijacking](https://www.upguard.com/blog/dll-hijacking) where running the application in the wrong directory causes unintentional code execution.

Reply:
You may open a new issue since this PR is closed.

Reply:
Thank you for the encouragement. I have opened https://github.com/fontforge/fontforge/pull/5515.